tests/test_target.py

```diff
@@ -14,35 +15,69 @@
 # limitations under the License.
 #
 
+""" Module for testing different targets. """
+
 import os
-import shutil
-import tempfile
 from unittest import TestCase
-
-from devlib import LocalLinuxTarget
-
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget
+from devlib.utils.misc import get_random_string
 
 class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self, target):
+        """ Test Target.read_tree_values_flat() """
+        method_name = self.__class__.test_read_multiline_values.__qualname__
 
-    def test_read_multiline_values(self):
         data = {
             'test1': '1',
             'test2': '2\n\n',
             'test3': '3\n\n4\n\n',
         }
 
-        tempdir = tempfile.mkdtemp(prefix='devlib-test-')
+        dirname = os.path.join(target.working_directory,
```

Comment:
looks like `target` is used but never defined

Reply:
`target` is defined in the new line #28.

Reply:
I dropped [07dd146](https://github.com/ARM-software/devlib/commit/07dd14616dc8d1ac8d2658d156baf1183924e53e) for now. So, `test/test_target.py` only has copyright change.
But, `target` was defined as input arg of `test_read_multiline_values()`.